src/plugins/workspace/server/utils.ts

```diff
@@ -0,0 +1,13 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import crypto from 'crypto';
+
+/**
+ * Generate URL friendly random ID
+ */
+export const generateRandomId = (size: number) => {
```

Comment:
The workspaceId will be prepend to url in the following PR, so a uuid with a length of 32 will be a little bit too large for a url path or query, so we choose to make it shorter by using this method.

Reply:
> The workspaceId will be prepended to url in the following PR, so a uuid with a length of 32 will be a little bit too large for a url path or query, so we choose to make it shorter by using this method.

That is pretty standard for the application. For example: 
https://playground.opensearch.org/app/dashboards#/view/7adfa750-4c81-11e8-b3d7-01146121b73d?_g=(filters:!(),refreshInterval:(pause:!f,value:900000),time:(from:now-24h,to:now))&_a=(description:'Analyze%20mock%20flight%20data%20for%20OpenSearch-Air,%20Logstash%20Airways,%20OpenSearch%20Dashboards%20Airlines%20and%20BeatsWest',filters:!(),fullScreenMode:!f,options:(hidePanelTitles:!f,useMargins:!t),query:(language:kuery,query:''),timeRestore:!t,title:'%5BFlights%5D%20Global%20Flight%20Dashboard',viewMode:view)`.

The url is pretty large and is what we utilize for global and app state so this can get even longer. I would be worried about potential conflicts. Custom generate ID functions tend to not actually be as random as they imply so it would be a lot safer to defer to OpenSearch to index and provide you with a verified unique ID.

If the length of the URL can be shorten utilize the short URL function
<img width="1439" alt="Screenshot 2023-09-27 at 11 32 14 PM" src="https://github.com/opensearch-project/OpenSearch-Dashboards/assets/12060704/d7a6154a-7026-4319-aa92-b17581ce01ac">.

Or you can also make titles unique ids as well. I believe index patterns are the same way.

Reply:
> (10000 / 256^6)

It's likely less than that:
https://nodejs.org/api/crypto.html#cryptorandombytessize-callback

> cryptographically strong pseudorandom data

So it will likely be based on a random seed value set by the OS.

> [https://playground.opensearch.org/w/${uuid_for_current_workspace}/app/dashboards],

Apologies, I wasn't aware of this. I believe this might have a larger implication to the application to how the application operates and will probably increase the scope of this project significantly.

Few examples on the top of my head:
* OSD starts up and discovers what plugins that are installed. It then takes the bundles for plugins and pushes under url/app/pluginName. So the current navigation system will have to be reworked. Because it will depend on url/app/pluginName to make the appropriate HTTP calls to download the bundles.
* History will have to be considered as well. Parts of the application and external plugins can be relying on the URL to handle history so if only parts of the application have the workspace route then they can get in a bad state.
* State management is managed by the URL via `_a` (for app) and `_g` so this might be impacted and you will have to modify existing OSD URL storage functions as some functions might be slicing up the route based on `url/{base_path}/app/plugin`
* APIs get hosted under url/api/pluginName

> make it more user friendly for user to see which routes they are viewing

Likely the URLs are obfuscated anyways depending on the size of your screen (mobile users). And if the goal is to be user friendly I think I would just go all the way and utilize the title as the unique id. 

**tl;dr**: So overall, I think that might add a lot of scope creep to add that and should be a proposal. To the point where I wouldn't want to consider that as a reason to keep in shorten Ids when we could add a new state like `_w`

Reply:
However things are difference between index patterns because workspace's title can be updated after a workspace has been created.

As for prepending workspaceId into our url path, there will be a RFC to discuss the options and we can forget that on this PR.

So what about we using a larger size, for example 12 maybe, as our workspaceId? As a length of 32 is too long for a url, especially when user copy a object detail page url, there will be a length of 32 to be used for workspaceId and a length of 32 to be used for object Id.

Reply:
I don't think using a 6 character ID doesn't make big difference than a 32 bytes UUID. and agree with @kavilla that if you really want to make it friendly, you can use user specified friendly id as the workspace document id instead using auto generated ones.

And regarding prepending the workspace id in the url path, I think Kibana namespace does it, but do we compare other options like using query parameter?

Reply:
@SuZhou-Joe any thinking around this?

Reply:
For generation method of workspace id, we have a [RFC](https://github.com/opensearch-project/OpenSearch-Dashboards/issues/5201) to discuss.

For workspace id in the url path, we do compare some options in the [RFC](https://github.com/opensearch-project/OpenSearch-Dashboards/issues/5243) we can use and in the end, prepending that into OSD path seems more reasonable. But this topic should not be discussed in this PR.